server/include/triton/developer_tools/server_wrapper.h

```diff
@@ -50,445 +47,45 @@ class InferRequest;
 struct ResponseParameters;
 class TraceManager;
 
-using TensorAllocMap = std::unordered_map<
-    std::string,
-    std::tuple<const void*, size_t, TRITONSERVER_MemoryType, int64_t>>;
-
-//==============================================================================
-/// Structure to hold logging options for setting 'ServerOptions'.
-///
-struct LoggingOptions {
-  // The range of VerboseLevel is [0, INT_MAX].
-  enum class VerboseLevel : int { OFF = 0, MIN = 1, MAX = INT_MAX };
-  enum class LogFormat { DEFAULT, ISO8601 };
-
-  LoggingOptions();
-
-  LoggingOptions(
-      const VerboseLevel verbose, const bool info, const bool warn,
-      const bool error, const LogFormat& format, const std::string& log_file);
-
-  // Verbose logging level. Default is OFF.
-  VerboseLevel verbose_;
-  // Enable or disable info logging level. Default is true.
-  bool info_;
-  // Enable or disable warn logging level. Default is true.
-  bool warn_;
-  // Enable or disable error logging level. Default is true.
-  bool error_;
-  // The format of logging. For "DEFAULT", the log severity (L) and
-  // timestamp will be logged as "LMMDD hh:mm:ss.ssssss". For "ISO8601", the
-  // log format will be "YYYY-MM-DDThh:mm:ssZ L". Default is 'DEFAULT'.
-  LogFormat format_;
-  // Logging output file. If specified, log outputs will be saved to this file.
-  // If not specified, log outputs will stream to the console. Default is an
-  // empty string.
-  std::string log_file_;  // logging output file
-};
-
-//==============================================================================
-/// Structure to hold metrics options for setting 'ServerOptions'.
-/// See here for more information:
-/// https://github.com/triton-inference-server/server/blob/main/docs/user_guide/metrics.md.
-struct MetricsOptions {
-  MetricsOptions();
-
-  MetricsOptions(
-      const bool allow_metrics, const bool allow_gpu_metrics,
-      const bool allow_cpu_metrics, const uint64_t metrics_interval_ms);
-
-  // Enable or disable metrics. Default is true.
-  bool allow_metrics_;
-  // Enable or disable GPU metrics. Default is true.
-  bool allow_gpu_metrics_;
-  // Enable or disable CPU metrics. Default is true.
-  bool allow_cpu_metrics_;
-  // The interval for metrics collection. Default is 2000.
-  uint64_t metrics_interval_ms_;
-};
-
-//==============================================================================
```

Comment:
With moving all these objects to  `generic_server_wrapper.h` file, I think maybe we'll need to update the README and point users to read `generic_server_wrapper.h` as well.